.github/workflows/dapp.yml

```
@@ -3,21 +3,34 @@ name: Dapp
 on:
   push:
     branches:
-      # TODO: Run only on master after we're fully migrated from Circle CI
-      - "rfc-18/**"
       - master
-    # TODO: paths:
+    paths-ignore:
+      - "docs/**"
+      - "infrastructure/**"
   pull_request:
-    branches:
-      # TODO: Run on all branches or only on master (to be decided) 
-      # after we're fully migrated from Circle CI
-      - "rfc-18/**"
-      - master
-    # TODO: paths:
   workflow_dispatch:
 
 jobs:
-  build-and-publish:
+  dapp-detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      path-filter: ${{ steps.filter.outputs.path-filter }}
+    steps:
+    - uses: actions/checkout@v2
+      if: github.event_name == 'pull_request' 
+    - uses: dorny/paths-filter@v2
+      if: github.event_name == 'pull_request' 
+      id: filter
+      with:
+        filters: |
+          path-filter:
+            - './!((docs|infrastructure)/**)'
+
+  dapp-build-and-publish:
+    needs: dapp-detect-changes
+    if: |
+      github.event_name != 'pull_request'
+        && needs.client-detect-changes.outputs.path-filter == 'true'
```

Shouldn't this be OR?
```suggestion
        || needs.client-detect-changes.outputs.path-filter == 'true'
```

---

Good catch, fixed in 0d24bef23a3b0a3d007b9f4c5279d815f9f1a018.